src/briefcase/bootstraps/toga.py

```diff
@@ -60,6 +60,11 @@ def pyproject_table_linux(self):
         return """\
 requires = [
     "toga-gtk~=0.4.7",
+    # PyGObject 3.52.1 enforces a requirement on libgirepository-2.0-dev. This library
+    # isn't available on Debian 12/Ubuntu 22.04. Enforce an upper pin that prevents that
+    # release (or newer) from being selected. If you don't need to support older
+    # Debian/Ubuntu releases, you can likely remove this pin. See beeware/toga#3143.
+    "pygobject < 3.52.1",
```

Comment:
FWIW, the _only_ PyGObject release less than `3.52.1` that works with `toga >= 0.4.7` is `3.50.0`. Prior versions of pygobject require gbulb and later versions have this new libgirepository-2.0 dependency.

Reply:
Sure - but that will get picked by the version bound math. `pygobject >= 3.50.0, < 3.52.1` will always resolve to 3.50.0.